components/pill/__tests__/pill.browser-test.jsx

```diff
@@ -0,0 +1,318 @@
+			expect(this.wrapper.prop('role')).to.equal('button');
+		});
+
+		it('renders label as a link', function () {
+			const anchor = this.wrapper.find('.slds-pill__action');
+			expect(anchor.exists()).to.be.true;
+			expect(anchor.prop('title')).to.equal(LABEL_TITLE);
+			const label = this.wrapper.find('.slds-pill__label');
+			expect(label.text()).to.equal(LABEL);
+		});
+
+		it('renders remove button', function () {
+			const removeButton = this.wrapper.find('.slds-pill__remove');
+			expect(removeButton.exists()).to.be.true;
+		});
+
+		it('renders assistive text in remove button', function () {
+			const removeButton = this.wrapper.find('.slds-pill__remove');
+			expect(removeButton.exists()).to.be.true;
+			const assistiveText = removeButton.find('.slds-assistive-text');
+			expect(assistiveText.exists()).to.be.true;
+			expect(assistiveText.text()).to.equal('Remove assistive text');
+		});
+
+		it('focuses and blurs', function () {
+			this.wrapper.instance().focus();
+			const pill = this.wrapper.find('.slds-pill');
+			expect(pill.matchesElement(document.activeElement)).to.be.equal(true, 'Pill was not focused');
+			expect(onFocus.calledOnce).to.be.true;
+			this.wrapper.instance().blur();
+			expect(pill.matchesElement(document.activeElement)).to.be.equal(false, 'Pill was not blurred');
+			expect(onBlur.calledOnce).to.be.true;
+		});
+
+		it('reponds to link clicks', function () {
+			const pillLink = this.wrapper.find('.slds-pill');
+			expect(pillLink.getNode()).to.exist;
+			Simulate.click(pillLink.getNode());
+			expect(onClick.calledOnce).to.be.true;
+		});
+
+		it('responds to remove clicks', function () {
+			const removeButton = this.wrapper.find('.slds-pill__remove');
+			expect(removeButton.getNode()).to.exist;
+			Simulate.click(removeButton.getNode());
+			expect(onRemove.calledOnce).to.be.true;
+		});
+	});
+
+	describe('Linked With Href', () => {
+		const HREF = 'https://www.salesforce.com';
+
+		beforeEach(mountComponent(
+			<SLDSPill
+				labels={{
+					label: LABEL
+				}}
+				href={HREF}
+			/>
+		));
+
+		afterEach(unmountComponent);
+
+		it('uses href', function () {
+			expect(this.wrapper.hasClass('slds-pill_link')).to.be.true;
+			const anchor = this.wrapper.find('.slds-pill__action');
+			expect(anchor.exists()).to.be.true;
+			expect(anchor.prop('href')).to.equal(HREF);
+		});
+	});
+
+	describe('Link style on', () => {
+		beforeEach(mountComponent(
+			<SLDSPill
+				labels={{
+					label: LABEL
+				}}
+				link
+			/>
+		));
+
+		afterEach(unmountComponent);
+
+		it('forces link style', function () {
+			expect(this.wrapper.hasClass('slds-pill_link')).to.be.true;
+			const anchor = this.wrapper.find('.slds-pill__action');
+			expect(anchor.exists()).to.be.true;
+		});
+	});
+
+	describe('Link style off', () => {
+		const onClick = sinon.stub();
+
+		beforeEach(mountComponent(
+			<SLDSPill
+				labels={{
+					label: LABEL
+				}}
+				onClick={onClick}
+				link={false}
```

Comment:
I intentionally did not want to introduce variants since I don't see those specified clearly in SLDS for the pill. Correct me if this is a common practice for DSR.
Otherwise, this implementation assumes that a normal way to specify whether the pill should appear as a linked pill or not is by specifying `onClick` callback: if it is specified, then the pill is a linked pill.
This does not fit with the prior private implementation of the pill since it does want `onClick`, but at the same time does not use the link style. 
Essentially this `link` property here is a way to override the normal behavior.

Reply:
I've read the requirements you are referring to, but I'm don't think they apply here. The logic around `link` property has 3 cases, which correspond to the values: `true`, `false`, `undefined`. There is no default value specified for `link`. Although I can change the way these 3 cases are structured and avoid use of `=== false` in favor of `===undefined`, if it makes anything clearer in any way...

Reply:
Yeah, I'd like to disconnect having an event mean changing markup / a UX pattern and believe a variant pattern would be helpful, since that typically means a significant markup/ UX pattern change. This one being is it a link or an option--but it can't be both. I think 16 components/sub-components have a `variant` prop. I'll try to work on some documentation about this. This is helpful.

DSR is a process...and doesn't always align with the `variant/theme/state/modifier` in SLDS. I've asked about some of those classifications and SLDS doesn't always have their ducks aligned (although they are better than DSR's various components 😉), so we have to think about what we want to do as engineers.

On a side note, the base UX pattern should probably be the default for all props, that is `link=true`. It's most likely to be used in isolation, option will always be in a listbox and the listbox can force the option on to the pill if it needs to.

Reply:
I'll change this such that `link` will control styling only and `onClick` will control whether the `role` is added to `<span/>` or not.
It does not make sense to me to add the variant here since the various related styling options such as `link`, `bare` & `error` are not mutually exclusive: you may want to have `link` combine with `error` etc.
One way to improve the component props could be to have one property, which a component consumer would use to specifying all of "styling" properties. Let me know if  DSR has an established pattern for that.